shell/platform/darwin/ios/framework/Source/FlutterViewController.mm

```diff
@@ -156,15 +157,22 @@ - (instancetype)init {
 
 - (void)sharedSetupWithProject:(nullable FlutterDartProject*)project
                   initialRoute:(nullable NSString*)initialRoute {
+  auto engine = fml::scoped_nsobject<FlutterEngine>{[[FlutterEngine alloc]
+                initWithName:@"io.flutter"
+                     project:project
+      allowHeadlessExecution:self.engineAllowHeadlessExecution]};
+
+  if (!engine) {
```

Comment:
I think this is ok but maybe this adds maintenance cost since we have to check everywhere that we respect this new contract and we should test all of these. Could we just assert in the engine init? There's no real point limping around further if the engine can't run.

Reply:
Umm I suppose the thinking here is that the user's not plugged in and will never see console logs by launching from home screen? Ya, never mind.

Reply:
> Could we just assert in the engine init?

We cannot assert as this will cause existing applications to crash when running on iOS 14. We limp here by displaying the placeholder.

Reply:
Ya, your point about add-to-app makes sense, SG